modules/social_features/social_group/social_group.module

```diff
@@ -2157,7 +2157,7 @@ function social_group_views_query_alter(ViewExecutable $view, QueryPluginBase $q
     return;
   }
   $account = \Drupal::currentUser()->getAccount();
-  if ($account->isAuthenticated()) {
+  if (!$account->isAuthenticated()) {
```

Comment:
Indeed checking for `if ($account->isAuthenticated())` creates a bug. After some investigation I think that we can change a bit the approach on social_group.module to something like this:

```
  $account = \Drupal::currentUser()->getAccount();

  if ($account->isAnonymous()) {
    // Get all hidden groups.
    $hidden_groups = \Drupal::entityTypeManager()
      ->getStorage('group')
      ->loadByProperties([
        'type' => 'closed_group',
      ]);

    // Remove all closed groups from /all-groups page.
    $ids = array_keys($hidden_groups);
    if ($ids) {
      /** @var \Drupal\views\Plugin\views\query\Sql $query */
      $query->addWhere('group_membership', 'groups_field_data.id', $ids, 'NOT IN');
    }
  }
```

We still need to check and remove closed_groups from anonymous users on that view ;) and reverting the entire commit will revert part of a bigger fix.

This should do the trick.

Can you please adjust your PR and let us know if this works as intended?